examples/example-tls/src/main/java/io/grpc/examples/helloworldtls/HelloWorldClientS2A.java

```diff
@@ -0,0 +1,114 @@
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import java.io.File;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.net.ssl.SSLException;
+import javax.annotation.Nonnull;
+
+// TODO: change flags to use com.google.common.flags
+
+/**
+ * Establishes a TLS 1.3 connection with a greeter service.
+ *
+ */
+public class HelloWorldClientS2A {
+    private static final Logger logger = Logger.getLogger(HelloWorldClientS2A.class.getName());
+    private final ManagedChannel channel;
+    private GreeterGrpc.GreeterBlockingStub blockingStub;
+
+    private static SslContext buildSslContext(String clientCertChainFilePath,
+                                              String clientPrivateKeyFilePath,
+                                              String trustCertCollectionFilePath) throws SSLException {
+      checkNotNull(clientCertChainFilePath, "clientCertChainFilePath should not be Null");
+      checkNotNull(clientPrivateKeyFilePath, "clientPrivateKeyFilePath should not be Null");
+      checkNotNull(trustCertCollectionFilePath, " trustCertCollectionFilePath should not be Null");
+      return GrpcSslContexts.forClient()
+            .trustManager(new File(trustCertCollectionFilePath))
+            .keyManager(new File(clientCertChainFilePath), new File(clientPrivateKeyFilePath))
+            .protocols("TLSv1.3")
+            .build();
+    }
+
+    /**
+     * Constructs client for accessing greeter server using the existing channel.
+     */
+    private HelloWorldClientS2A(ManagedChannel channel, GreeterGrpc.GreeterBlockingStub blockingStub) {
+        this.channel = channel;
+        this.blockingStub = blockingStub;
+    }
+
+    public void shutdown() throws InterruptedException {
+        channel.shutdown().awaitTermination(5, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Says hello to a greeter server.
+     */
+    private void greet(String name) {
+        checkNotNull(name, "name should not be Null");
+        logger.info("Will try to greet " + name + " ...");
+        HelloRequest request = HelloRequest.newBuilder().setName(name).build();
+        HelloReply response = blockingStub.sayHello(request);
+        logger.info("Greeting: " + response.getMessage());
+    }
+
+    private static HelloWorldClientS2A create(String host, int port, SslContext sslContext) throws SSLException {
+        checkNotNull(host, "host should not be Null");
+        checkNotNull(port,"port should not be Null");
+        checkNotNull(sslContext, "SslContext should not be Null");
+        ManagedChannel channel = NettyChannelBuilder.forAddress(host,port)
+                        .sslContext(sslContext)
+                        .build();
+        return new HelloWorldClientS2A(channel, GreeterGrpc.newBlockingStub(channel));
+    }
+
+    /**
+     * Sends a {@code HelloRequest} to a greeter service.
+     *
+     * args:
```

Comment:
For each parameter, please use the following notation.

`@param arg[0] contains the server address in the form host:port.`